### PR TITLE
Removed unnecessary files from META-INF going to final bundle

### PR DESCRIPTION
**Pull request by thalescm**

### CODE REVIEW

**Comment by kageiit** [comment]
cc @styurin

**Comment by styurin** [comment]
I'm not sure this is a reasonable approach to hard the list of entries. Why these entries are there and can they be stripped out before? What if there are more entries that needs to be removed?

**Comment by kageiit** [comment]
I think the ApkBuilder from aosp does this implicitly today for apks in buck. What alternative approach would you suggest @styurin 

Can we follow something like https://github.com/facebook/buck/blob/master/src/com/facebook/buck/android/ApkBuilderStep.java#L178 ?

**Comment by styurin** [comment]
I was thinking about having an argument on a rule that would list entries to remove, but these details may look very low level for top level rule.

**Comment by kageiit** [comment]
> I was thinking about having an argument on a rule that would list entries to remove, but these details may look very low level for top level rule.

I think such a list would be very hard to maintain. I think what we wanted here is an include list rather than an exclude list. But having a list would be too low level as you noticed. This current change mirrors what the apk builder does in buck. Maybe we can add documentation to the code to make it clearer why we choose these entries only?

**Comment by styurin** [comment]
> Maybe we can add documentation to the code to make it clearer why we choose these entries only?

Sounds good.

**Comment by thalescm** [comment]
@styurin Se if you think that is enough please :)

**Comment by thalescm** [comment]
Couldn't find any relationship of the CI failures with this PR, so they seem unrelated to me

**Comment by styurin** [comment]
Sorry for late comments, I forgot to post them

**Comment by thalescm** [comment]
@styurin if you can see if this makes it more readable pls

File: src/com/facebook/buck/android/AabBuilderStep.java [hunk 1]
```diff
@@ -419,6 +418,41 @@ private void packageFile(
...
+  private boolean isEntryPackageable(ZipEntry entry) {
+    return isDirectoryEntryPackageable(entry) || isFileEntryPackageable(entry);
+  }
+
+  private boolean isDirectoryEntryPackageable(ZipEntry entry) {
+    return entry.isDirectory() && ApkBuilder.checkFolderForPackaging(entry.getName());
+  }
+
+  private boolean isFileEntryPackageable(ZipEntry entry) {
+    String entryName = entry.getName();
```
Comment (bobyangyf) [inline src/com/facebook/buck/android/AabBuilderStep.java]: don't need this. Just refer to entry.getName() in both places

**Comment by thalescm** [comment]
@bobyangyf done 👍 
@styurin how did the import go?

**Pull request closed without merging**